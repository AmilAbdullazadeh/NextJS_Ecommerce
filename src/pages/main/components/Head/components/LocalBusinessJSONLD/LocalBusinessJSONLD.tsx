import React from 'react';

import { GIT_REPO_URL, SITE_IMAGE_URL, SITE_NAME, SITE_URL } from '@core';

const LocalBusinessJSONLD: React.FC = () => {
    const JSON_LD = {
        '@context': 'http://schema.org',
        '@type': 'LocalBusiness',
        name: SITE_NAME,
        description: 'An ecommerce site selling technology products',
        url: SITE_URL,
        image: [SITE_IMAGE_URL],
        sameAs: [GIT_REPO_URL],
        contactPoint: {
            contactType: 'PR',
            email: 'sprgoulas@gmail.com',
            hoursAvailable: ['Mo-Fri 09:00-21:00', 'Sa-Su 11:00-13:00'],
        },
        founder: 'Spyros Goulas',
        foundingDate: '2021-5-15',
        numberOfEmployees: '1',
        slogan: `${SITE_NAME}, it has it all!`,
        currenciesAccepted: 'EUR, USD, BTC',
        paymentAccepted: 'Cash, Credit Card, Cryptocurrency',
    };

    return (
        <script
            type="application/ld+json"
            dangerouslySetInnerHTML={{ __html: JSON.stringify(JSON_LD) }}
        ></script>
    );
};

export default LocalBusinessJSONLD;
