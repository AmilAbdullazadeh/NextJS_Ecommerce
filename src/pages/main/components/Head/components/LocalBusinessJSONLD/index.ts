export { default as LocalBusinessJSONLD } from './LocalBusinessJSONLD';
