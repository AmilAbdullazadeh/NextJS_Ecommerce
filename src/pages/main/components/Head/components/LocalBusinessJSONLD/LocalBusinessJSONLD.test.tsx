import React from 'react';

import { render } from '@testUtils';

import LocalBusinessJSONLD from './LocalBusinessJSONLD';

describe('LocalBusinessJSONLD', () => {
    it('renders correctly', () => {
        const {
            container: { firstChild },
        } = render(<LocalBusinessJSONLD />);

        expect(firstChild).toMatchSnapshot();
    });
});
