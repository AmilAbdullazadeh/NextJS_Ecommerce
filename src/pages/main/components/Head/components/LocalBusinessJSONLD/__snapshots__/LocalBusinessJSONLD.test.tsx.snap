// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocalBusinessJSONLD renders correctly 1`] = `
<script
  type="application/ld+json"
>
  {"@context":"http://schema.org","@type":"LocalBusiness","name":"PDP project","description":"An ecommerce site selling technology products","url":"http://localhost:3000","image":["http://localhost:3000/ecommerce.jpg"],"sameAs":["https://github.com/sgoulas/pdpProject"],"contactPoint":{"contactType":"PR","email":"sprgoulas@gmail.com","hoursAvailable":["Mo-Fri 09:00-21:00","Sa-Su 11:00-13:00"]},"founder":"Spyros Goulas","foundingDate":"2021-5-15","numberOfEmployees":"1","slogan":"PDP project, it has it all!","currenciesAccepted":"EUR, USD, BTC","paymentAccepted":"Cash, Credit Card, Cryptocurrency"}
</script>
`;
